openwisp_controller/config/base/vpn.py

```diff
@@ -948,6 +966,52 @@ def _auto_ip(self):
                 return
         self.ip = self.vpn.subnet.request_ip()
 
+    def _generate_zt_identity(self):
+        try:
+            result = subprocess.run(
+                'zerotier-idtool generate',
+                shell=True,
+                check=True,
+                # in seconds
+                timeout=5,
+                capture_output=True,
+            )
+        except (CalledProcessError, TimeoutExpired) as exc:
+            err = getattr(exc, 'stderr', None)
+            # In case of timeout
+            if err is None:
+                err = exc
+            err_msg = f'Unable to generate zerotier identity secret, Error: {err}'
+            raise ZeroTierIdentityGenerationError(err_msg)
+        return result.stdout.decode('utf-8')
+
+    def _auto_zt_identity_secret(self):
+        if not self.zt_identity_secret and self.vpn._is_backend_type('zerotier'):
+            self.zt_identity_secret = self._generate_zt_identity()
+            self.member_id = self.zt_identity_secret[:10]
```

Comment:
I would re-write this as following for better readability and consistency 

```suggestion
        if not self.vpn._is_backend_type('zerotier') or self.zt_identity_secret:
            return 
        # If there's an existing ZeroTier VpnClient for this device, the re-use the zt_identity_secret
        existing_zt_client = self.__class__.objects.only('zt_identity_secret').exclude(id=self.id).filter(config_id=self.config_id, zt_identity_secret__isnull=False).first()
        if existing_ztn_client: 
            self.zt_identity_secret = existing_ztn_client.zt_identity_secret
        else:
            self.zt_identity_secret = self._generate_zt_identity()
```